### PR TITLE
Makes several new kinds of alter table statements involving auto_increment columns work correctly

### PR DESCRIPTION
**Pull request by zachmu**
Also: makes several kinds of ALTER TABLE statements with multiple clauses more lenient in their error checking than MySQL. These statements will now succeed instead of being rejected.

Fixes https://github.com/dolthub/dolt/issues/6218

### CODE REVIEW

**Comment by zachmu** [comment]
I need to do another pass on CREATE TABLE statements that contain similar constructions, I doubt they work properly yet

**Comment by zachmu** [comment]
> I need to do another pass on CREATE TABLE statements that contain similar constructions, I doubt they work properly yet

Nevermind, I added tests of this and it does work correctly

**Review by fulghum — APPROVED** [review]
Looks good! 🚢

File: enginetest/enginetests.go [hunk 1]
```diff
@@ -5415,86 +5415,6 @@ func TestAddDropPks(t *testing.T, harness Harness) {
 	})
 }
 
-func TestAddAutoIncrementColumn(t *testing.T, harness Harness) {
```
[the remark this answers is not in the document]
Comment (zachmu) [reply enginetest/enginetests.go:5418]: I have that fixed in another PR

File: sql/plan/alter_table.go [hunk 2]
```diff
@@ -727,6 +728,15 @@ func NewModifyColumn(database sql.Database, table *UnresolvedTable, columnName s
 	}
 }
 
+func (m *ModifyColumn) MarkAddColumn() {
+	m.partOfAddColumn = true
+}
```
[the remark this answers is not in the document]
Comment (zachmu) [reply sql/plan/alter_table.go:733]: Might be leftover from an earlier approach, will take a look

File: sql/analyzer/validate_create_table.go [hunk 3]
```diff
@@ -191,6 +199,37 @@ func resolveAlterColumn(ctx *sql.Context, a *Analyzer, n sql.Node, scope *plan.S
...
+		for _, col := range n.Columns {
+			keyedColumns[col.Name] = true
+		}
+	case plan.IndexAction_Drop:
+		for _, col := range n.Columns {
+			delete(keyedColumns, col.Name)
+		}
+	}
+
+	return keyedColumns
```
[the remark this answers is not in the document]
Comment (zachmu) [reply sql/analyzer/validate_create_table.go:232]: This is true

**Pull request merged**